### PR TITLE
Handle no email for user in database in Edit profile view

### LINKED ISSUE #545: Users that don't have an entry in db cannot update profile
**Describe the bug**
The update profile view doesn't load and a completely blank page is displayed.

**To Reproduce**
1. I have an account that doesn't have a user entry in the db (due to a previous issue with the prediction of the safe addresses when validating an account)
2. Click on 'Edit Profile'
3. See error: The edit profile view doesn't load and a completely blank page is displayed. And in Network tab of the browser: **Request**: `https://api.circles.garden/api/users/<safe-address>/email/`, **Response**: `404 Not Found`

**Expected behavior**
Load the page with both "username" and "email" fields empty.

### PR DESCRIPTION
**Pull request by mikozet**
Closes https://github.com/CirclesUBI/circles-myxogastria/issues/545

### CODE REVIEW

File: src/views/EditProfile.js [hunk 1]
```diff
@@ -375,9 +375,15 @@ const EditProfile = () => {
 
   useEffect(() => {
     (async () => {
-      const userEmail = await core.user.getEmail(safe.currentAccount);
-      setEmailInput(userEmail);
-      setCurrentUserEmail(userEmail);
+      try {
+        const userEmail = await core.user.getEmail(safe.currentAccount);
```
Comment (louilinn) [inline src/views/EditProfile.js]: Why not only 

`
const userEmail = await core.user.getEmail(safe.currentAccount) || '';`

I don't think we should log this as an error?

**Review by louilinn — CHANGES_REQUESTED** [review]

File: src/views/EditProfile.js (hunk 1 above)
Comment (mikozet) [reply src/views/EditProfile.js]: I think catch is more secure here as the line which you have mentioned will not register if it is an error object I believe. 
Generally we should log all type of errors as a good practice as it is easier to debug in future

Comment (louilinn) [reply src/views/EditProfile.js]: Yes I agree we should log errors but I think for fetching user data that doesnt exist we shouldnt log it as an error. For example when an account doesnt have a usename we don't log it as an error - then the address is displayed instead. Why should a missing email be an error? 
You can see in username.js how we deal with no avatar or no username.
I think we want to make it possible to remove email at some point and also for data erasure requests. These accounts shouldn't then have an email and it is then incorrect to log an error. 
If there is an error with the core call (unlike returning null which isn't an error) there will still be an error? Then we should have try catch around everything in the app which isnt what this issue requests

File: src/views/EditProfile.js [hunk 2]
```diff
@@ -375,9 +375,19 @@ const EditProfile = () => {
 
   useEffect(() => {
     (async () => {
-      const userEmail = await core.user.getEmail(safe.currentAccount);
-      setEmailInput(userEmail);
-      setCurrentUserEmail(userEmail);
+      if (safe.currentAccount) {
+        const userEmail = await core.user.getEmail(safe.currentAccount);
+        if (userEmail.status !== 404) {
```
Comment (louilinn) [inline src/views/EditProfile.js]: I don't see getEmail ever returning an object with a status attribute?
https://github.com/CirclesUBI/circles-core/blob/e838131b64ae82be2b30fa12e4cb6f2a042f53f0/src/user.js#L270

**Review by louilinn — COMMENTED** [review]
Would you like to elaborate on why you didn't like the suggestion I proposed?

`const userEmail = await core.user.getEmail(safe.currentAccount) || '';`

**Comment by mikozet** [comment]
> 



> Would you like to elaborate on why you didn't like the suggestion I proposed?
> 
> `const userEmail = await core.user.getEmail(safe.currentAccount) || '';`

I just doesn't feel like it would solve the issue here. If you get 404 from that query how would it save it from crashing?

**Review by llunaCreixent — DISMISSED** [review]
Tested with dockers and correctly reproduced the error with the main branch, and it's solved woth this branch.
I created an account and removed the user entry in the docker db to reproduce the user case.

Good work!

**Comment by louilinn** [comment]
I don't feel that I can comment further on code quality as I cannot test it (without having such an account). So since lluna is the one to test I leave it to her to also review the code quality and if all these if else statements are necessary. And in that case approve the PR @llunaCreixent

**Comment by llunaCreixent** [comment]
> I don't feel that I can comment further on code quality as I cannot test it (without having such an account). So since lluna is the one to test I leave it to her to also review the code quality and if all these if else statements are necessary. And in that case approve the PR @llunaCreixent 

@louilinn you can test locally by removing the entry in the docker database after creating the account.

**Comment by llunaCreixent** [comment]
The solution that @louilinn suggested works better because it doesn't produce an "Uncaught Promise" error.
When the api is down, we get `400` error, When not found we receive `404`. But in any case we display empty email.

**Review by llunaCreixent — DISMISSED** [review]

**Comment by mikozet** [comment]
> The solution that @louilinn suggested works better because it doesn't produce an "Uncaught Promise" error.
> When the api is down, we get `400` error, When not found we receive `404`. But in any case we display empty email.

When we get 404 it is an object so how we will get an empty email? It is a valid response I believe

**Review by louilinn — APPROVED** [review]

**Pull request merged**